cmd/nitro/nitro.go

```diff
@@ -741,55 +771,59 @@ func mainImpl() int {
-	MetricsServer          genericconf.MetricsServerConfig `koanf:"metrics-server"`
-	PProf                  bool                            `koanf:"pprof"`
-	PprofCfg               genericconf.PProf               `koanf:"pprof-cfg"`
-	Init                   conf.InitConfig                 `koanf:"init"`
-	Rpc                    genericconf.RpcConfig           `koanf:"rpc"`
-	BlocksReExecutor       blocksreexecutor.Config         `koanf:"blocks-reexecutor"`
-	EnsureRollupDeployment bool                            `koanf:"ensure-rollup-deployment" reload:"hot"`
+	Conf                                  genericconf.ConfConfig          `koanf:"conf" reload:"hot"`
+	Node                                  arbnode.Config                  `koanf:"node" reload:"hot"`
+	Execution                             gethexec.Config                 `koanf:"execution" reload:"hot"`
+	Validation                            valnode.Config                  `koanf:"validation" reload:"hot"`
+	ParentChain                           conf.ParentChainConfig          `koanf:"parent-chain" reload:"hot"`
+	Chain                                 conf.L2Config                   `koanf:"chain"`
+	LogLevel                              string                          `koanf:"log-level" reload:"hot"`
+	LogType                               string                          `koanf:"log-type" reload:"hot"`
+	FileLogging                           genericconf.FileLoggingConfig   `koanf:"file-logging" reload:"hot"`
+	Persistent                            conf.PersistentConfig           `koanf:"persistent"`
+	HTTP                                  genericconf.HTTPConfig          `koanf:"http"`
+	WS                                    genericconf.WSConfig            `koanf:"ws"`
+	IPC                                   genericconf.IPCConfig           `koanf:"ipc"`
+	Auth                                  genericconf.AuthRPCConfig       `koanf:"auth"`
+	GraphQL                               genericconf.GraphQLConfig       `koanf:"graphql"`
+	Metrics                               bool                            `koanf:"metrics"`
+	MetricsServer                         genericconf.MetricsServerConfig `koanf:"metrics-server"`
+	PProf                                 bool                            `koanf:"pprof"`
+	PprofCfg                              genericconf.PProf               `koanf:"pprof-cfg"`
+	Init                                  conf.InitConfig                 `koanf:"init"`
+	Rpc                                   genericconf.RpcConfig           `koanf:"rpc"`
+	BlocksReExecutor                      blocksreexecutor.Config         `koanf:"blocks-reexecutor"`
+	EnsureRollupDeployment                bool                            `koanf:"ensure-rollup-deployment" reload:"hot"`
+	ExecutionNode                         bool                            `koanf:"execution-node"`
+	ConsensusExecutionInSameProcessUseRPC bool                            `koanf:"consensus-execution-in-same-process-use-rpc"`
```

Comment:
let's instead just use the Node.ExecutionRPCClient.URL field:
* If it's empty (default) - create an execution node and use it directly.
* if it's "self" or "self-auth" - create an execution node and connect to it via rpc (still connect to it directly for fullExecutionNode)